diff --git a/reports/opendmarc-reports.in b/reports/opendmarc-reports.in
index 2dce0e4..344148c 100755
--- a/reports/opendmarc-reports.in
+++ b/reports/opendmarc-reports.in
@@ -823,8 +823,6 @@ foreach (@$domainset)
 		    $uri->opaque eq "")
 		{
 			print STDERR "$progname: can't parse reporting URI for domain $domain\n";
-			unlink($zipfile);
-			unlink($repfile);
 			next;
 		}
 
@@ -871,9 +869,6 @@ foreach (@$domainset)
 			{
 				print STDERR "$progname: unknown URI scheme in '$repuri' for domain $domain\n";
 			}
-
-			unlink($zipfile);
-			unlink($repfile);
 			next;
 		}
 		# send/post report
@@ -930,8 +925,6 @@ foreach (@$domainset)
 			if (!open($zipin, $zipfile))
 			{
 				print STDERR "$progname: can't read zipped report for $domain: $!\n";
-				unlink($zipfile);
-				unlink($repfile);
 				next;
 			}
 
@@ -1009,8 +1002,6 @@ foreach (@$domainset)
 		else
 		{
 			print STDERR "$progname: unsupported reporting URI scheme " . $uri->scheme . " for domain $domain\n";
-			unlink($zipfile);
-			unlink($repfile);
 			next;
 		}
 	}
