Index: opendmarc-1.3.1/libopendmarc/opendmarc_policy.c
===================================================================
--- opendmarc-1.3.1.orig/libopendmarc/opendmarc_policy.c
+++ opendmarc-1.3.1/libopendmarc/opendmarc_policy.c
@@ -495,7 +495,7 @@ opendmarc_policy_store_dkim(DMARC_POLICY
 	 * If so and if we have not already found
 	 * a best match, make this the temporary best match.
 	 */
-	if (opendmarc_policy_check_alignment(dp, pctx->from_domain, pctx->adkim))
+	if (opendmarc_policy_check_alignment(dp, pctx->from_domain, pctx->adkim) == 0)
 	{
 		if (pctx->dkim_domain != NULL)
 		{
