diff --git a/configure b/configure
index bb72d9a..eaaf6c2 100755
--- a/configure
+++ b/configure
@@ -15041,7 +15041,7 @@
 #
 # Generate files
 #
-ac_config_files="$ac_config_files Makefile contrib/Makefile contrib/init/Makefile contrib/init/generic/Makefile contrib/init/redhat/Makefile contrib/init/redhat/opendmarc contrib/rddmarc/Makefile contrib/service/Makefile contrib/service/opendmarc.service contrib/spec/Makefile contrib/spec/opendmarc.spec db/Makefile docs/Makefile libopendmarc/Makefile libopendmarc/tests/Makefile libopendmarc/tests/testfiles/Makefile libopendmarc/docs/Makefile opendmarc/Makefile opendmarc/opendmarc.conf.5 opendmarc/opendmarc.8 opendmarc/opendmarc-check.8 opendmarc/tests/Makefile reports/Makefile reports/opendmarc-expire reports/opendmarc-expire.8 reports/opendmarc-import reports/opendmarc-import.8 reports/opendmarc-importstats.8 reports/opendmarc-params reports/opendmarc-params.8 reports/opendmarc-reports reports/opendmarc-reports.8"
+ac_config_files="$ac_config_files Makefile contrib/Makefile contrib/init/Makefile contrib/init/generic/Makefile contrib/init/redhat/Makefile contrib/init/redhat/opendmarc contrib/rddmarc/Makefile contrib/service/Makefile contrib/service/opendmarc.service contrib/spec/Makefile contrib/spec/opendmarc.spec db/Makefile docs/Makefile libopendmarc/Makefile libopendmarc/tests/Makefile libopendmarc/tests/testfiles/Makefile libopendmarc/docs/Makefile opendmarc/Makefile opendmarc/opendmarc.conf.5 opendmarc/opendmarc.8 opendmarc/opendmarc-check.8 opendmarc/tests/Makefile reports/Makefile reports/opendmarc-expire reports/opendmarc-expire.8 reports/opendmarc-import reports/opendmarc-import.8 reports/opendmarc-importstats reports/opendmarc-importstats.8 reports/opendmarc-params reports/opendmarc-params.8 reports/opendmarc-reports reports/opendmarc-reports.8"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -16092,6 +16092,7 @@
     "reports/opendmarc-expire.8") CONFIG_FILES="$CONFIG_FILES reports/opendmarc-expire.8" ;;
     "reports/opendmarc-import") CONFIG_FILES="$CONFIG_FILES reports/opendmarc-import" ;;
     "reports/opendmarc-import.8") CONFIG_FILES="$CONFIG_FILES reports/opendmarc-import.8" ;;
+    "reports/opendmarc-importstats") CONFIG_FILES="$CONFIG_FILES reports/opendmarc-importstats" ;;
     "reports/opendmarc-importstats.8") CONFIG_FILES="$CONFIG_FILES reports/opendmarc-importstats.8" ;;
     "reports/opendmarc-params") CONFIG_FILES="$CONFIG_FILES reports/opendmarc-params" ;;
     "reports/opendmarc-params.8") CONFIG_FILES="$CONFIG_FILES reports/opendmarc-params.8" ;;
diff --git a/reports/opendmarc-importstats b/reports/opendmarc-importstats
deleted file mode 100755
index 839a871..0000000
--- a/reports/opendmarc-importstats
+++ /dev/null
@@ -1,26 +0,0 @@
-#!/bin/sh
-##
-## Copyright (c) 2012, The Trusted Domain Project.  All rights reserved.
-##
-## opendmarc-importstats -- import opendmarc output to MySQL
-##
-## This is intended to be used via a crontab.  If import is successful,
-## this code exits quietly so there's no output.  If it fails, it does
-## "ls -l" on the temporary file, so that cron generates mail to whever
-## ran the job.
-
-## setup
-statsdb="/var/tmp/dmarc.dat"
-# OPENDMARC_PASSWORD="password"; export OPENDMARC_PASSWORD
-
-if [ -s $statsdb ]
-then
-	mv $statsdb ${statsdb}.OLD.$$
-
-	if opendmarc-import < ${statsdb}.OLD.$$
-	then
-		rm ${statsdb}.OLD.$$
-	else
-		ls -l ${statsdb}.OLD.$$
-	fi
-fi
diff --git a/reports/opendmarc-importstats.in b/reports/opendmarc-importstats.in
new file mode 100755
index 0000000..3a28ee3
--- /dev/null
+++ b/reports/opendmarc-importstats.in
@@ -0,0 +1,27 @@
+#!/bin/sh
+##
+## Copyright (c) 2012, The Trusted Domain Project.  All rights reserved.
+##
+## opendmarc-importstats -- import opendmarc output to MySQL
+##
+## This is intended to be used via a crontab.  If import is successful,
+## this code exits quietly so there's no output.  If it fails, it does
+## "ls -l" on the temporary file, so that cron generates mail to whever
+## ran the job.
+
+## setup
+statsdb="`grep ^HistoryFile @sysconfdir@/opendmarc.conf | sed 's/^HistoryFile\s\+//'`"
+[ -z "$statsdb" ] && exit 0
+# OPENDMARC_PASSWORD="password"; export OPENDMARC_PASSWORD
+
+if [ -s "$statsdb" ]
+then
+	mv "$statsdb" "${statsdb}.OLD.$$"
+
+	if @sbindir@/opendmarc-import < ${statsdb}.OLD.$$
+	then
+		rm "${statsdb}.OLD.$$"
+	else
+		ls -l "${statsdb}.OLD.$$"
+	fi
+fi
