Index: opendmarc-1.3.2/opendmarc/opendmarc.c
===================================================================
--- opendmarc-1.3.2/opendmarc/opendmarc.c.orig	2016-07-20 01:50:38.948795907 -0400
+++ opendmarc-1.3.2/opendmarc/opendmarc.c	2016-07-20 01:54:27.408805245 -0400
@@ -2620,9 +2620,13 @@
 				char *mode;
 
 				if (mode == DMARC_POLICY_SPF_ORIGIN_HELO)
-					mode = "helo:
+				{
+					mode = "helo:";
+				}
 				else
+				{
 					mode = "mailfrom";
+				}
 
 				syslog(LOG_INFO,
 				       "%s: SPF(%s): %s %s",
