diff --git a/opendmarc/opendmarc.c b/opendmarc/opendmarc.c
index fa9d1a4..60c7f3e 100644
--- a/opendmarc/opendmarc.c
+++ b/opendmarc/opendmarc.c
@@ -2754,8 +2754,7 @@ mlfi_eom(SMFICTX *ctx)
 	dmarcf_dstring_printf(dfc->mctx_histbuf, "align_spf %d\n", align_spf);
 
 	/* prepare human readable policy string for later processing */
-	ostatus = opendmarc_get_policy_token_used(cc->cctx_dmarc);
-	switch (ostatus == DMARC_USED_POLICY_IS_SP ? sp : p)
+	switch (opendmarc_get_policy_token_used(cc->cctx_dmarc) == DMARC_USED_POLICY_IS_SP ? sp : p)
 	{
 	  case DMARC_RECORD_P_QUARANTINE:
 		apolicy = "quarantine";
