diff --git a/opendmarc/opendmarc.c b/opendmarc/opendmarc.c
index 29f3f93..c5594fc 100644
--- a/opendmarc/opendmarc.c
+++ b/opendmarc/opendmarc.c
@@ -2750,12 +2750,17 @@ mlfi_eom(SMFICTX *ctx)
 	{
 	  case DMARC_POLICY_ABSENT:		/* No DMARC record found */
 	  case DMARC_FROM_DOMAIN_ABSENT:	/* No From: domain */
-	  case DMARC_POLICY_NONE:		/* Accept and report */
 		aresult = "none";
 		ret = SMFIS_ACCEPT;
 		result = DMARC_RESULT_ACCEPT;
 		break;
 
+	  case DMARC_POLICY_NONE:		/* Alignment failed, but policy is none: */
+		aresult = "fail";		/* Accept and report */
+		ret = SMFIS_ACCEPT;
+		result = DMARC_RESULT_ACCEPT;
+		break;
+
 	  case DMARC_POLICY_PASS:		/* Explicit accept */
 		aresult = "pass";
 		ret = SMFIS_ACCEPT;
