diff --git a/reports/opendmarc-reports.in b/reports/opendmarc-reports.in
index ba0de76..e81aaf0 100755
--- a/reports/opendmarc-reports.in
+++ b/reports/opendmarc-reports.in
@@ -23,6 +23,7 @@ use IO::Compress::Zip qw(zip);
 use POSIX;
 use MIME::Base64;
 use Net::SMTP;
+use Time::Local;
 
 require DBD::@SQL_BACKEND@;
 
@@ -52,7 +53,7 @@ my $zipfile;
 
 my $zipin;
 
-my $now;
+my $now = time();
 
 my $repstart;
 my $repend;
@@ -276,7 +277,12 @@ if (!defined($dbuser))
 	}
 }
 
-if (!defined($interval))
+if (defined($interval) && $daybound)
+{
+	print STDERR "$progname: WARN: --day overrides --interval\n";
+}
+
+if (!defined($interval) || $daybound)
 {
 	$interval = $def_interval;
 }
@@ -287,7 +293,7 @@ $keepfiles = ($testmode == 1) ? 1 : $keepfiles;
 
 if ($verbose)
 {
-	print STDERR "$progname: started at " . localtime() . "\n";
+	print STDERR "$progname: started at " . localtime($now) . "\n";
 }
 
 my $dbi_dsn = "DBI:" . $dbscheme . ":database=" . $dbname .
@@ -322,8 +328,6 @@ if ($use_utc)
 # Select domains on which to report
 #
 
-$now = time();
-
 if ($verbose >= 2)
 {
 	print STDERR "$progname: selecting target domains\n";
@@ -491,9 +495,22 @@ foreach (@$domainset)
 		next;
 	}
 
+	if ($daybound)
+	{
+		my ($sec,$min,$hour,$mday,$mon,$year,$wday,$yday,$isdst) = localtime($now - $interval);
+		$repstart = timelocal(0, 0, 0, $mday, $mon, $year);
+		($sec,$min,$hour,$mday,$mon,$year,$wday,$yday,$isdst) = localtime($now);
+		$repend = timelocal(0, 0, 0, $mday, $mon, $year);
+	}
+	else
+	{
+		$repstart = $now - $interval;
+		$repend = $now;
+	}
+
 	# construct the temporary file
-	$repfile = $repdom . "!" . $domain . "!" . $lastsent . "!" . time() . ".xml";
-	$zipfile = $repdom . "!" . $domain . "!" . $lastsent . "!" . time() . ".zip";
+	$repfile = $repdom . "!" . $domain . "!" . $repstart . "!" . $repend . ".xml";
+	$zipfile = $repdom . "!" . $domain . "!" . $repstart . "!" . $repend . ".zip";
 	if (!open($tmpout, ">", $repfile))
 	{
 		print STDERR "$progname: can't create report file for domain $domain\n";
@@ -531,39 +548,6 @@ foreach (@$domainset)
  		else		{ $spolicystr = "unknown"; }
  	}
 
-	if ($daybound)
-	{
-		$dbi_s = $dbi_h->prepare("SELECT UNIX_TIMESTAMP(MIN(date)), UNIX_TIMESTAMP(MAX(date)) FROM messages WHERE from_domain = ? AND DATE(date) >= DATE(FROM_UNIXTIME(?)) AND DATE(date) < DATE(FROM_UNIXTIME(?))");
-	}
-	else
-	{
-		$dbi_s = $dbi_h->prepare("SELECT UNIX_TIMESTAMP(MIN(date)), UNIX_TIMESTAMP(MAX(date)) FROM messages WHERE from_domain = ? AND UNIX_TIMESTAMP(date) > ? AND UNIX_TIMESTAMP(date) <= ?");
-	}
-
-	if (!$dbi_s->execute($domainid, $lastsent, $now))
-	{
-		print STDERR "$progname: can't extract begin/end times for domain $domain: " . $dbi_h->errstr . "\n";
-		$dbi_s->finish;
-		$dbi_h->disconnect;
-		exit(1);
-	}
-
-	$repstart = 0;
-	$repend = $now;
-
-	while ($dbi_a = $dbi_s->fetchrow_arrayref())
-	{
-		if (defined($dbi_a->[0]))
-		{
-			$repstart = $dbi_a->[0];
-		}
-		if (defined($dbi_a->[1]))
-		{
-			$repend = $dbi_a->[1];
-		}
-	}
-
-	$dbi_s->finish;
 
 	print $tmpout "<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\n";
 	print $tmpout "<feedback>\n";
@@ -596,7 +580,7 @@ foreach (@$domainset)
 		$dbi_s = $dbi_h->prepare("SELECT messages.id, ipaddr.addr, messages.disp, d1.name, d2.name, messages.spf, messages.align_spf, messages.align_dkim FROM messages JOIN ipaddr ON messages.ip = ipaddr.id JOIN domains d1 ON messages.from_domain = d1.id JOIN domains d2 ON messages.env_domain = d2.id WHERE messages.from_domain = ? AND messages.date > FROM_UNIXTIME(?) AND messages.date <= FROM_UNIXTIME(?)");
 	}
 
-	if (!$dbi_s->execute($domainid, $lastsent, $now))
+	if (!$dbi_s->execute($domainid, $repstart, $repend))
 	{
 		print STDERR "$progname: can't extract report for domain $domain: " . $dbi_h->errstr . "\n";
 		$dbi_s->finish;
@@ -956,7 +940,7 @@ foreach (@$domainset)
 	if ($doupdate)
 	{
 		$dbi_s = $dbi_h->prepare("UPDATE requests SET lastsent = FROM_UNIXTIME(?) WHERE domain = ?");
-		if (!$dbi_s->execute($now, $domainid))
+		if (!$dbi_s->execute($repend, $domainid))
 		{
 			print STDERR "$progname: can't update last sent time for domain $domain: " . $dbi_h->errstr . "\n";
 			$dbi_s->finish;
